x/fees/keeper/msg_server.go

```diff
@@ -94,75 +113,103 @@ func (k Keeper) RegisterFee(
 	return &types.MsgRegisterFeeResponse{}, nil
 }
 
-// CancelFee deletes the Fee of a given contract
-func (k Keeper) CancelFee(
+// UpdateFee updates the withdraw address of a given Fee
+func (k Keeper) UpdateFee(
 	goCtx context.Context,
-	msg *types.MsgCancelFee,
-) (*types.MsgCancelFeeResponse, error) {
+	msg *types.MsgUpdateFee,
+) (*types.MsgUpdateFeeResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
-	if !k.isEnabled(ctx) {
-		return nil, sdkerrors.Wrapf(types.ErrInternalFee, "fees module is not enabled")
+
+	params := k.GetParams(ctx)
+	if !params.EnableFees {
+		return nil, types.ErrFeesDisabled
 	}
 
-	contractAddress := common.HexToAddress(msg.ContractAddress)
-	deployerAddress, found := k.GetDeployer(ctx, contractAddress)
+	contract := common.HexToAddress(msg.ContractAddress)
+	fee, found := k.GetFee(ctx, contract)
 	if !found {
-		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "contract %s is not registered", msg.ContractAddress)
+		return nil, sdkerrors.Wrapf(
+			types.ErrFeesContractNotRegistered, "contract %s is not registered", msg.ContractAddress,
+		)
 	}
 
-	if msg.DeployerAddress != deployerAddress.String() {
-		return nil, sdkerrors.Wrapf(sdkerrors.ErrUnauthorized, "%s is not the contract deployer", msg.DeployerAddress)
+	if msg.DeployerAddress != fee.DeployerAddress {
+		return nil, sdkerrors.Wrapf(
+			types.ErrFeesDeployerIsNotEOA, "%s is not the contract deployer", msg.DeployerAddress,
```

Comment:
q: why `ErrFeesDeployerIsNotEOA`? the account type is not related to the CREATE address derivation

Reply:
I think I mixed it up here. So the deployer address is always the original deployer?